lib/Conversion/TorchToLinalg/Uncategorized.cpp

```diff
@@ -930,10 +930,14 @@ static Value createLinalgPayloadCalculationForElementwiseOp(
   }
   if (auto atenToDtype = dyn_cast<AtenToDtypeOp>(op)) {
     Value input = payloadArgs[0];
+    Type srcOriginalDtype =
+        converter->convertType(atenToDtype.getSelf().getType())
+            .cast<RankedTensorType>()
+            .getElementType();
     Type dtype = converter->convertType(atenToDtype.getType())
                      .cast<RankedTensorType>()
                      .getElementType();
-    Value result = convertScalarToDtype(b, loc, input, dtype);
+    Value result = convertScalarToDtype(b, loc, input, dtype, srcOriginalDtype);
```

Comment:
The change will lower `aten.to.dtype(ui8) -> f32` to `arith.uitofp`, instead of `arith.sitofp`.